src/main/java/sg/ncl/MainController.java

```diff
@@ -5016,4 +5026,182 @@ private TeamQuota extractTeamQuotaInfo(String responseBody) {
+                error = objectMapper.readValue(responseBody, MyErrorResource.class);
+                ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
+                switch (exceptionState) {
+                    case PASSWORD_RESET_REQUEST_TIMEOUT_EXCEPTION:
+                        log.warn("Error for class member password reset for user {}: password reset key timeout", uid);
+                        newClassMemberPasswordResetForm.setErrMsg("Password reset request timed out. Please request a new reset email.");
+                        break;
+
+                    case INVALID_USERNAME_EXCEPTION:
+                        log.warn("Error for class member password reset for user {}: invalid username", uid);
+                        newClassMemberPasswordResetForm.setErrMsg("You must enter a valid first and last name.");
+                        break;
+
+                    case INVALID_PASSWORD_EXCEPTION:
+                        log.warn("Error for class member password reset for user {}: password is invalid", uid);
+                        newClassMemberPasswordResetForm.setErrMsg("You must supply a valid password.");
+                        break;
+
+                    case INVALID_CREDENTIALS_EXCEPTION:
+                        log.warn("Error for class member password reset for user {}: uid and key do not match", uid);
+                        newClassMemberPasswordResetForm.setErrMsg(ERR_SERVER_OVERLOAD);
+                        break;
+
+                    case CREDENTIALS_NOT_FOUND_EXCEPTION:
+                        log.warn("Error for class member password reset for user {}: credentials not found", uid);
+                        newClassMemberPasswordResetForm.setErrMsg("Key is not valid");
+                        break;
+
+                    default:
+                        log.warn("Adding members by emails : sio or deterlab adapter connection error");
+                        newClassMemberPasswordResetForm.setErrMsg(ERR_SERVER_OVERLOAD);
+                }
+
+                return NEW_MEMBER_RESET_PASSWORD;
+            } catch (IOException e) {
+                throw new WebServiceRuntimeException(e.getMessage());
+            }
+        }
+
+
+        log.info("New class member password reset successful for user {}", uid);
+        newClassMemberPasswordResetForm.setSuccessMsg("Password reset is successful");
+        return "password_reset_success";
+    }
+
+
+
+    // when class member clicks password reset key link in the email
+    @GetMapping(path = "/newMemberResetKey", params = {"uid"})
+    public String resetKey(@NotNull @RequestParam("uid") final String uid) {
+
+        HttpEntity<String> request = createHttpEntityHeaderOnlyNoAuthHeader();
+        restTemplate.setErrorHandler(new MyResponseErrorHandler());
+        try {
+            ResponseEntity responseEntity = restTemplate.exchange(properties.newMemberResetKey(uid), HttpMethod.PUT, request, String.class);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "responseEntity". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "responseEntity" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1481)